syntax = "proto3";

package maskDetection;

message Alert {

  string event_time = 1;

  message Device {
    string type = 1;
    string guid = 2;
    string enrolled_on = 3;
  }
  Device created_by = 2;

  message Location {
    float longitude = 1;
    float latitude = 2;
  }

  Location location = 3;

  message Model {
    string name = 1;
    string guid = 2;
    float threshold = 3;
  }

  Model face_detection_model = 4;
  Model mask_classifier_model = 5;

  float probability = 6;

  message Image {

    message Size {
      int32 width = 1;
      int32 height = 2;
    }
    string format = 1;
    Size size = 2;
    bytes data = 3;
  }
  Image image = 7;
}
